source/elements/oneMKL/source/domains/dft/descriptor.rst

```diff
@@ -346,7 +351,84 @@ corresponding configuration values, see :ref:`onemkl_dft_enum_config_param`.
 
 **Descriptor class member table:** :ref:`onemkl_dft_descriptor_member_table`
 
+.. _onemkl_dft_descriptor_set_workspace:
 
+set_workspace
++++++++++++++
+
+Set the workspace for when ``config_param::WORKSPACE_PLACEMENT`` is set to ``config_value::WORKSPACE_EXTERNAL``.
+
+.. rubric:: Description
+
+This function sets the workspace to use when computing DFTs for when an
+external workspace is set. 
+This function may only be called after the descriptor has been committed.
+The provided workspace must be equal to or larger than the required 
+workspace size obtained by calling ``descriptor<prec, dom>::get_value(config_param::WORKSPACE_EXTERNAL_BYTES_REQUIRED, &workspaceBytes)``.
+
+A descriptor where ``WORKSPACE_EXTERNAL`` is specified is not a valid descriptor 
+for compute calls until this function has been successfully called.
+
+The type of workspace must match the compute calls for which it is used.
+That is, if the workspace is provided as a ``sycl::buffer``, the compute
+calls must also use ``sycl::buffer`` for their arguments. Likewise, a USM
+allocated workspace must only be used with USM compute calls.
+Failing to do this will result in an invalid descriptor for compute calls.
+
+If the workspace provided is a USM allocation shared amongst multiple kernels,
+it must be ensured that the kernels sharing this workspace do not use it simultaneously.
+
+This function can be called on committed descriptors where the workspace placement
+is not ``config_value::WORKSPACE_EXTERNAL``. The provided workspace may or may not
+be used in compute calls. However, the aforementioned restrictions will still apply.
+
+Calling this function multiple times on a committed descriptor is undefined behavior.
```

Comment:
Why do we need this restriction? Is there any harm in allowing to set the workspace multiple times or to switch the workspace between compute functions?

Reply:
My understanding is that their probably isn't any harm, but I don't know how things work under-the-hood for closed-source FFT libraries. For example a valid optimization might be to only write twiddle factors to device memory once, rather than on every compute call. It also avoid having to discuss edge cases where I suspect this wouldn't be valid.

Reply:
Oh right, I was wondering whether the DFT domain could benefit from keeping the workspace reserved. We had similar discussions about the external workspace for the sparse blas domain and the conclusion is that we should allow users to re-use the external workspace as they wish between computes if that means losing some performance. I think it would make sense to do the same for the DFT domain and document that `WORKSPACE_EXTERNAL` can prevent some optimizations because of this. I would be curious to hear the opinion of Helen or Raphael on this.
Do you know if cuFFT and rocFFT mention anything about the restrictions on the external workspace?

Reply:
By reserved I mean do we allow users to write to the external workspace between 2 compute functions. Whichever the decision we should clarify whether this is allowed. As you said currently with the current wording users could either re-use the workspace for other descriptor or other kernels. I think that makes sense but in that case I don't see the point of having `set_workspace` undefined behavior.
I assume that cuFFT also allows to write on the external workspace between 2 computes if they don't mention any restriction.

Reply:
I've reworded it in [073c506](https://github.com/oneapi-src/oneAPI-spec/pull/509/commits/073c506d3f9dc468d6f464adae4278ee8f8ff2ad) - do you think this is clearer?

Reply:
Yes, I have a preference for that version, thanks.